src/wp-includes/rest-api/endpoints/class-wp-rest-autosaves-controller.php

```diff
@@ -247,7 +247,11 @@ public function create_item( $request ) {
 			return $autosave_id;
 		}
 
-		$autosave = get_post( $autosave_id );
+		$autosave = $this->get_post( $autosave_id );
+		if ( is_wp_error( $autosave ) ) {
```

Comment:
I think this check is still needed because `wp_update_post()` above can return 0.